cmd/object-api-options.go

```diff
@@ -66,9 +66,8 @@ func getDefaultOpts(header http.Header, copySource bool, metadata map[string]str
 	if crypto.S3.IsRequested(header) || (metadata != nil && crypto.S3.IsEncrypted(metadata)) {
 		opts.ServerSideEncryption = encrypt.NewSSE()
 	}
-	if v, ok := header[xhttp.MinIOSourceProxyRequest]; ok {
-		opts.ProxyHeaderSet = true
-		opts.ProxyRequest = strings.Join(v, "") == "true"
+	if _, ok := header[xhttp.MinIOSourceProxyRequest]; ok {
+		opts.ProxyRequest = true
```

Comment:
```suggestion
		_, opts.ProxyRequest = header[xhttp.MinIOSourceProxyRequest]
```

you wanted to simplify :-)

Reply:
Good catch. 🙂